src/expr.cpp

```diff
@@ -5737,11 +5740,14 @@ llvm::Value *VectorMemberExpr::GetValue(FunctionEmitContext *ctx) const {
 }
 
 int VectorMemberExpr::getElementNumber() const {
-    int elementNumber = lIdentifierToVectorElement(identifier[0]);
-    if (elementNumber == -1) {
+    int idx = lIdentifierToVectorElement(identifier[0]);
+    if (idx == -1) {
         Error(pos, "Vector element identifier \"%s\" unknown.", identifier.c_str());
+    } else if (exprVectorType != nullptr && idx >= exprVectorType->GetElementCount()) {
```

Comment:
I am wondering if we need to check this earlier, for example, during the construction of `VectorMemberExpr` with an incorrect subscript. In other words, does ISPC report errors for all other possible uses, e.g., like this: `sizeof(col.w)`?

Reply:
Right, other error messages are constructed here: https://github.com/ispc/ispc/pull/3427/files#diff-080c8b0513a71a985e8647431edc401ac005e46ce039b830be9dc20fd073ff1fR5815

Reply:
> I am not convinced calling `Error` in a constructor is a great idea. 

It looks like there are already quite a lot of errors reported from constructors but I don't have a strong opinion on this.

> I agree that sizeof(col.w) was an issue ...
> PS: I added the `sizeof(col.w)` in the tests.

Thanks for fixing this!